### PR TITLE
feat: added governance proposal details page

### PR DESCRIPTION
**Pull request by ChiragPansuriya-iView**
design updates as per figma 

Screenshot 2024-03-01 at 9 09 40 AM

Mobile 
Screenshot 2024-03-14 at 12 28 32 PM

### CODE REVIEW

File: packages/hooks/governance/useProposal.ts
Comment (clegirar) [inline packages/hooks/governance/useProposal.ts:1]: The `useProposal` and `useProposals` can be confusing i think, just one letter differ from each other, so maybe `useGetProposal` and `useGetAllProposals` can be better ? It's a proposition (im not sure about the `Get` in it, but it means that you fetch a data)

File: packages/screens/Governance/GovernanceProposal/GovernanceVoteHeader/components/VoteTimeDetailsBox.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+              fontSemibold13,
+              {
+                marginTop: layout.spacing_x0_25,
+                color: secondaryColor,
+              },
+            ]}
+          >
+            {`${submit_time.slice(0, 10)} ${submit_time.slice(11, 16)} UTC`}
+          </BrandText>
+        </View>
```
Comment (clegirar) [inline packages/screens/Governance/GovernanceProposal/GovernanceVoteHeader/components/VoteTimeDetailsBox.tsx:61]: This component cannot be extract ? I think that is the same pattern below no ? If yes extract it and the file will be easier to read too. I think this little component can stay in the file

**Review by clegirar — CHANGES_REQUESTED** [review]
Just little things to fix 👍 
Ill check the design this afternoon

File: packages/hooks/governance/useProposal.ts
Comment (ChiragPansuriya-iView) [reply packages/hooks/governance/useProposal.ts:1]: noted

**Comment by omniwired** [comment]
Mobile styles need some love.
image

**Review by omniwired — CHANGES_REQUESTED** [review]
mobile styles

File: packages/screens/Governance/GovernanceProposal/GovernanceProposalScreen.tsx [hunk 2]
```diff
@@ -9,21 +9,39 @@ import { GovernanceVoteHeader } from "./GovernanceVoteHeader/GovernanceVoteHeade
...
+        );
+        const data = await res.json();
+
+        setProposal(data.proposal);
+      } catch (err) {
+        console.error(err);
+      }
+    };
+    effect();
+  }, [id, selectedNetworkId]);
```
Comment (n0izn0iz) [inline packages/screens/Governance/GovernanceProposal/GovernanceProposalScreen.tsx:44]: use react-query for queries please

File: packages/screens/Governance/GovernanceScreen.tsx [hunk 3]
```diff
@@ -7,19 +7,34 @@ import { GovernanceBox } from "../../components/GovernanceBox/GovernanceBox";
...
+        );
+        const data = await res.json();
+
+        setProposals(data.proposals.reverse());
+      } catch (err) {
+        console.error(err);
+      }
+    };
+    effect();
+  }, [selectedNetworkId]);
```
Comment (n0izn0iz) [inline packages/screens/Governance/GovernanceScreen.tsx]: react-query

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

**Comment by clegirar** [comment]
image

Just i have this behavior on mobile, on web, maybe you can make sure that the search input and `All states` button in one line.
And the `Reviews` text and the separator are overlapped, just fix that too

**Pull request merged**